proposals/NNNN-hlsl-loop-unroll-factor.md

```diff
@@ -0,0 +1,38 @@
+# HLSL Loop Unroll Factor
+
+* Proposal: [NNNN](NNNN-hlsl-loop-unroll-factor.md)
+* Author(s): [Farzon Lotfi](https://github.com/farzonl)
+* Sponsor: [Farzon Lotfi](https://github.com/farzonl)
+* Status: **Under Consideration**
+* Impacted Projects: DXC & Clang
+
+## Introduction
+
+This proposal seeks to unify the diverging behaviors between the agreed upon 
+[spec for HLSL loop unroll](https://github.com/microsoft/hlsl-specs/pull/263)
+and the current behavior of DXC.
+
+The new spec wants to treat the unroll factor as a hint to the compiler for 
+partial loop unroll. The current behavior in dxc is that the unroll factor 
+specifies the maximum number of times the loop is to execute. 
+
+The DXC behavior was determined to violate user expectations by overriding 
+the existing loop termination conditions. Further the DXC behavior diverges 
+from how both clang and openCL treat the loop unroll factor which would have 
+made our port of HLSL loop unroll a bunch special cases instead of just 
+syntactic sugar that could sit on top of the existing loop unroll 
+implementations that exist in LLVM.
+
+## Motivation
+The HLSL compilers transition to clang has resulted in a compat break between 
+the [HLSLS loop unroll implementation in clang](https://github.com/llvm/llvm-project/pull/93879)
+and the one in DXC. While there is an expectation that the new compiler will 
+not be fully compatible with the previous compilers, These compat breaks should
+ be minimized when possible.
+
+ ## Proposed solution
+In [HLSL 202X](0020-hlsl-202x-202y.md) the DXC implementation of the loop 
+unroll factor should no longer alter the  number of iteration executions. 
+Ideally the loop unroll factor will also become a compiler hint for partial 
+loop unroll. If this can not be done because of concerns of invalid DXIL 
+generation then the loop unroll factor as a feature should be removed from DXC.
```

Comment:
It would be good to clarify what "removed" means here. Are you thinking this would this be a hard error? Or just ignore the value and issue a warning.

Reply:
To lower the code impact it makes sense to follow standard deprecation procedures. Ignore the value and issue a warning makes the most sense.